test/functional/rpc_createmultisig.py

```diff
@@ -133,117 +84,162 @@ def check_addmultisigaddress_errors(self):
         pubs = [self.nodes[1].getaddressinfo(addr)["pubkey"] for addr in addresses]
         assert_raises_rpc_error(-5, "Bech32m multisig addresses cannot be created with legacy wallets", self.nodes[0].addmultisigaddress, 2, pubs, "", "bech32m")
 
-    def checkbalances(self):
-        node0, node1, node2 = self.nodes
-        self.generate(node0, COINBASE_MATURITY)
+    def test_multisig_script_limit(self, wallet_multi):
+        node1 = self.nodes[1]
+        pubkeys = self.pub
 
-        bal0 = node0.getbalance()
-        bal1 = node1.getbalance()
-        bal2 = node2.getbalance()
-        balw = self.wallet.get_balance()
+        self.log.info('Test legacy redeem script max size limit')
+        assert_equal(len(pubkeys), 20)
+        assert_raises_rpc_error(-8, "redeemScript exceeds size limit: 684 > 520", node1.createmultisig, 16, pubkeys, 'legacy')
 
-        height = node0.getblockchaininfo()["blocks"]
-        assert 150 < height < 350
-        total = 149 * 50 + (height - 149 - 100) * 25
-        assert bal1 == 0
-        assert bal2 == self.moved
-        assert_equal(bal0 + bal1 + bal2 + balw, total)
+        self.log.info('Test valid 16-20 multisig p2sh-legacy and bech32 (no wallet)')
+        self.do_multisig(nkeys=20, nsigs=16, output_type="p2sh-segwit", wallet_multi=None)
+        self.do_multisig(nkeys=20, nsigs=16, output_type="bech32", wallet_multi=None)
```

Comment:
do we need to cover m=21 here to catch the error in segwit multisig?

Reply:
yeah, good idea. Will add it.